classes/manager.php

```diff
@@ -304,69 +185,33 @@ public static function get_min_duration_for_reason(int $reason, bool $usecache =
      * @param float duration of the current profile
      * @return bool whether or not the profile should stored with the REASON_SLOW reason.
      */
-    public static function is_considered_slow(profile $profile): bool {
-        $durationms = $profile->get('duration') * 1000; // Convert to ms.
+    public function is_considered_slow(profile $profile): bool {
+        $duration = $profile->get('duration');
 
         // First, check against the overall minimum duration value to ensure it meets the
         // minimum required duration for the profile to be considered slow.
-        if ($durationms <= self::min_duration()) {
+        if ($duration <= $this->processor->get_min_duration()) {
             return false;
         }
 
         // If a min duration exists, it means the quota is filled, and only
         // profiles slower than the fastest stored profile should be stored.
-        $minduration = self::get_min_duration_for_reason(profile::REASON_SLOW);
-        if ($minduration && $durationms <= $minduration) {
+        $minduration = profile::get_min_duration_for_reason(profile::REASON_SLOW);
```

Comment:
Should this be a static reference? I can imagine a world where the "slow" duration varies between profile objects, and it seems like being static here is against the pattern of the rest of the commits

Reply:
This comment, and the next, address the same thing. So this response is for both. These functions were written by Kevin, so I avoid messing with them unless I have to. I moved them to profile because they are relevant to profiles (they access the profile database table). They are static because are aggregate functions. They do not refer to a specific item.